### DISCUSSION TITLE
Is there a simple way to shift from SQLIte to Postgres or MySQL?

### DISCUSSION THREAD

**Discussion by TheBig-O** [jovandeginste/workout-tracker#315] [Q&A] [answered]
I imagine that I could simply export everything from SQLite as individual tables and import them into Postgres, but I'm curious if there's a way to perform an sqldump from one and upload it to the other. 
I haven't decided if there is an advantage to shifting over, but I already have a few Postgres databases and a backup routine. So, one more wouldn't be a big change.

**Comment by TheBig-O** [comment, marked as answer]
The short answer appears to be no, there is no way to move easily. Instead, I used an IDE to extract the notes from all the workouts in the Sqlite version and matched them up with the imported GPX data in the Postgres version. Because all the files were retained in the "done" folder, doing this was relatively simple. Wat gets missed are the associations to equipment and route segments. Neither is a huge issue, but it would be great if I could figure out a simple way to convert an SQL dump to a format compatible with Postgres. (Many fields types are not compatible and need to be reassigned.)
One thing that doesn't work is PGLoader. Tried that, didn't work.

**Reply by jovandeginste** [reply]
Dit you find https://stackoverflow.com/questions/4581727/how-to-convert-sqlite-sql-dump-file-to-postgresql ?

**Reply by TheBig-O** [reply]
I missed the deferred constraints initially, but got things working well!
I'm much more comfortable with Postgres than SQLite. 
That was my reason for wanting to shift over.
